File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        
        HashMap<Integer, Integer> seen = new HashMap<>();
        seen.put(0,1);
        
        int sum=0;
        int count=0;

        for(int num:nums){
            sum+=num;
            int req=sum-k;

            if(seen.containsKey(req)){
                count+=seen.get(req);
            }
            seen.put(sum,seen.getOrDefault(sum,0)+1);
        }
        return count;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution(object):
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        count={0: 1}
        total,answer=0,0
        for num in nums:
            total+= num
            if total-k in count:
                answer += count[total-k]
            count[total] = count.get(total,0)+1
        return answer